drafts/client-interface.md

```diff
@@ -3,11 +3,150 @@
 This document describes the interface of a yet-to-be-built reference client
 library in pseudo code.
 
+The hydra client works similar to a browser. You point it to a URL. The URL becomes the current resource. 
+Then you interact with the controls of the resource.
 
-## Retrieve a resource
+The examples show how a cli application could use the hydra client for interaction with an API.
+
+## Find API entrypoint for website
+
+Hydra has a special mechanism to discover API entrypoints.
+
+```
+hydra.findEntrypoint(website)
+print hydra.representation 
+// the API entrypoint is now the current representation
```

Comment:
The resource that is returned could have a method to get the link to the ApiDocumentation.

Dietrich, what do you think about the proposal 

> to go through the comments and address them in whatever way you feel is appropriate. Adding a section "open questions" or something along those lines would work quite well for lots of them given that this is a working doc. We can then merge this and someone else can pick up one of the questions or make changes to the doc that we will then discuss separately.

I wouldn't like this to get stuck. Your approach (top-down) is completely fine. We don't need to work bottom-up.